openstudiocore/src/model/test/ThermalZone_GTest.cpp

```diff
@@ -809,3 +810,32 @@ TEST_F(ModelFixture, ThermalZone_ZoneControlHumidistat)
   }
 }
 
+TEST_F(ModelFixture, ThermalZone_RenderingColor) {
+  Model m;
+
+  ThermalZone thermalZone(m);
+  RenderingColor renderingColor(m);
+  EXPECT_TRUE(thermalZone.setRenderingColor(renderingColor));
+
+  EXPECT_EQ(1u, m.getModelObjects<RenderingColor>().size());
+
+  // Delete ThermalZone, since the color is referenced, it should be deleted too
+  EXPECT_EQ(3u, thermalZone.remove().size()); // Zone + ZoneHVACEquipmentList + Color
+  EXPECT_EQ(0u, m.getModelObjects<ThermalZone>().size());
+  EXPECT_EQ(0u, m.getModelObjects<RenderingColor>().size());
+
+
+  // Clone
+  ThermalZone thermalZone2(m);
+  RenderingColor renderingColor2(m);
+  EXPECT_TRUE(thermalZone2.setRenderingColor(renderingColor2));
```

Comment:
@jmarrec setRenderingColor should make sure that each color has 0 or 1 parent.  It should clone as needed?  This should also be done at version translation

Reply:
I'll implement that in all classes that have rendering colors in setRenderingColor.
It'll make it check if it already has a parent, in which case Warn and return false as discussed.